### ISSUE TITLE
Dependent pages can have empty titles

### ISSUE THREAD

**Issue by scott1702** [silverstripe/silverstripe-cms#2428]
Following on from https://github.com/silverstripe/silverstripe-cms/pull/2427. It's a fairly common pattern for CMS users to not add titles to things like content blocks.
Because of this, any blocks which link to a page and don't have titles appear as empty rows on the dependent pages table.
![image](https://user-images.githubusercontent.com/10215604/57493278-57be2800-7318-11e9-9709-9f28a6d712d2.png)

One way to fix this would be to use the singular_name and ID where a title doesn't exist:
![image](https://user-images.githubusercontent.com/10215604/57493297-6e647f00-7318-11e9-8219-d7696504af8b.png)

But there are other formats that this could be in e.g. always having the type of object but using ID where a title doesn't exist:
```
Home | Home Page | Click here to edit
#123 | Content block | Click here to edit
```

Keen to get feedback on a preferred format. /cc @silverstripeux

### Solution
- Use the singular name, i.e 'Untitled block'

**Comment by robbieaverill** [comment]
We discussed a little on Slack around using `getTitle()` which would return `#123` (ID) if no Title or Name is available which is nice out of the box, but without the context of what model type it is it would just be confusing. Perhaps we add another column to show the data model singular name to help clarify that.

**Comment by sachajudd** [comment]
All blocks should have a default name when you create them (we think @ScopeyNZ worked on this, correct me if I'm wrong). It should be e.g. “Untitled content block” or "Untitled file block" or something similar. Here are some [Designs](https://projects.invisionapp.com/dsm/silver-stripe/silver-stripe/asset/components/5b90499d4d45900010a7be60) indicating the titles. This could be a similar issue that's occurring here https://github.com/dnadesign/silverstripe-elemental/issues/574

**Comment by robbieaverill** [comment]
Note that this issue affects any DataObject in a project with an HTML field on it, not just elemental

**Comment by ScopeyNZ** [comment]
> All blocks should have a default name when you create them (we think @ScopeyNZ worked on this, correct me if I'm wrong).

Yeah in the editor you use when creating/modifying content blocks it will fill in "untitled block" in place of an empty title, but the title is actually still nothing.

**Comment by newleeland** [comment]
> Yeah in the editor you use when creating/modifying content blocks it will fill in "untitled block" in place of an empty title, but the title is actually still nothing.

Its would be really beneficial if these placeholder names (untitled block) were the actual names of the blocks.

**Comment by ScopeyNZ** [comment]
Okay - we can do that. This is a discussion for `silverstripe-elemental` though so please raise it there. I personally think it's nicer to have the titles actually blank, because we can display `untitled` with different styling (eg. italics) and it's likely a content author will want to completely redefine a title of an "untitled" block, so we're pre-empting them deleting the content of the field.

I think this issue makes sense to handle cases where titles are blank, and we shouldn't adjust elemental to fix this. Just my 2¢

**Comment by robbieaverill** [comment]
I completely agree with @ScopeyNZ. Any DataObject with a HTML field could be shown in this list, and you can't expect user code to always have a Title field on it. We need to adjust core (and the UI) to handle DataObject's that don't have a title, regardless of how elemental behaves currently.

**Comment by newleeland** [comment]
> I completely agree with @ScopeyNZ. Any DataObject with a HTML field could be shown in this list, and you can't expect user code to always have a Title field on it. We need to adjust core (and the UI) to handle DataObject's that don't have a title, regardless of how elemental behaves currently.

Myself and @ScopeyNZ came to an agreement over lunch, that its fine for a content block to have an empty title value, if there was no title, but show the user an "untitled block" placeholder title. i.e. how Blocks works currently. However, it would also be good to see these placeholder titles of "dependant page" and/or "Used on" in Files,
rather than: 
- using a cryptic ID or 
- leaving it empty (which may look broken to the user)

**Comment by robbieaverill** [comment]
So we're essentially talking about `Untitled [lowercase i18n_singular_name]` in terms of implementing this I think. It's good to use blocks as an example of this issue, but we shouldn't restrict ourselves to thinking about only blocks

**Comment by brynwhyman** [comment]
@scott1702 did you end up implementing something in a project to to resolve this issue?

**Comment by scott1702** [comment]
Yep we implemented as singular_name + ID so e.g. if a content element if it didn't have a title it would say `Content Element (#72)`

**Comment by ScopeyNZ** [comment]
Fixed in 4.4.x-dev: #2458
